### PR TITLE
feat: add organization mapping model to map external orgs code to internal organization

### PR DESCRIPTION
**Pull request by AfaqShuaib09**
This PR covers the following JIRA ticket:
#### [PROD-3199](https://2u-internal.atlassian.net/browse/PROD-3199)

This PR covers the creation of OrganizationMapping model to map the `external_org_codes` to internal `Organization` model.

Besides creating `OrganizationMapping` model, this change also fixes flaky tests in LMS Data loader. The tests were failing because of an additional OAuth call that happened during test suite setUp. This resulted in responses call count incrementing by 1 and failing the tests.

#### ADR
[ADR of product source](https://github.com/openedx/course-discovery/pull/3826)

### Testing
- Run migrations and verify the model is created
    ```bash
    ./manage.py migrate course_metadata
    ```
- Verify the model is created in the admin panel
- Run the tests
```bash
pytest course_discovery/apps/course_metadata/tests/test_models.py::OrganizationMappingTests --ds=course_discovery.settings.test_local
```

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @AfaqShuaib09!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: course_discovery/apps/course_metadata/models.py [hunk 1]
```diff
@@ -305,6 +305,30 @@ def save(self, *args, **kwargs):
             raise e
 
 
+class OrganizationMapping(models.Model):
+    """
+    Model to map an internal organization to the corresponding organization in an external product source.
+    organization: Internal organization
+    source: External product source
+    external_organization_code: Organization code in an external product source
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/models.py]: There is no need for the field names in docstring.

File: course_discovery/apps/course_metadata/models.py [hunk 2]
```diff
@@ -305,6 +305,30 @@ def save(self, *args, **kwargs):
...
+    """
+    organization = models.ForeignKey('Organization', models.CASCADE)
+    source = models.ForeignKey('Source', models.CASCADE)
+    external_organization_code = models.CharField(
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/models.py]: nit: organization_external_key is better suited and similar to key field in Organization.

**Review by DawoudSheraz — COMMENTED** [review]
In PR description,
- Add link to ADR PR where the info of this addition is present
- Flaky tests fixup

File: course_discovery/apps/course_metadata/models.py [hunk 3]
```diff
@@ -305,6 +305,27 @@ def save(self, *args, **kwargs):
             raise e
 
 
+class OrganizationMapping(models.Model):
+    """
+    Model to map external/third party organization codes to the internal organization
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/models.py]: nit: to organizations internal to Discovery.

**Review by DawoudSheraz — APPROVED** [review]

**Review by AliAdnanSohail — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@AfaqShuaib09 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/course-discovery&entry.867055334=https://github.com/openedx/course-discovery/pull/3828&entry.1484655318=https://github.com/AfaqShuaib09&entry.752974735=2023-03-06+20:07&entry.1917517419=2023-03-08+08:13&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**